### PR TITLE
Adds XHR and ViewerRender proxy modules

### LINKED ISSUE #11: Template rendering module
The rendering counterpart to #7. Reacts to `viewerRenderTemplate` messages and forwards them to the server endpoint for rendering templates.

### PR DESCRIPTION
**Pull request by fstanis**
Closes #11 and #12

### CODE REVIEW

File: src/framecontainer/rendering-modules/ViewerRenderProxy.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { FrameContainer } from '../FrameContainer';
+import { Messaging } from '@ampproject/viewer-messaging';
+import { TemplateProxyRequest, TemplateProxyResponse } from './ProxyRequest';
+import { Config } from '../../config';
+import { isValidURL, postJSON } from '../../util';
```
Comment (alabiaga) [inline src/framecontainer/rendering-modules/ViewerRenderProxy.ts]: where are these defined? I can't find them in this PR or currently existing. Maybe you forgot to add them?

File: src/framecontainer/rendering-modules/XHRProxy.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { FrameContainer } from '../FrameContainer';
+import { Messaging } from '@ampproject/viewer-messaging';
+import { ProxyRequest, ProxyResponse } from './ProxyRequest';
+import { Config } from '../../config';
+import { isValidURL, postJSON } from '../../util';
+
+/**
+ * Reacts to changes to the AMP document's height and resizes the iframe to
```
Comment (alabiaga) [inline src/framecontainer/rendering-modules/XHRProxy.ts]: Is the class description correct?

File: src/framecontainer/rendering-modules/ViewerRenderProxy.ts (hunk 1 above)
Comment (fstanis) [reply src/framecontainer/rendering-modules/ViewerRenderProxy.ts]: Sorry for the confusion: this PR is on top of #36 (just merged) that I split from it.

File: src/framecontainer/rendering-modules/XHRProxy.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * adjust to them.
+ */
+class XHRProxyImpl {
+  private readonly proxyURL: string;
+  private readonly messaging: Messaging;
+
+  constructor(frameContainer: FrameContainer) {
+    const config = frameContainer.getConfig();
+    this.proxyURL = config.xhrProxyURL || config.templateProxyURL || '';
```
Comment (alabiaga) [inline src/framecontainer/rendering-modules/XHRProxy.ts]: why is the templateProxyUrl config being used here?

File: src/framecontainer/viewerConfig.ts [hunk 4]
```diff
@@ -53,13 +53,20 @@ export const VIEWER_PARAMETERS: ViewerParameters = {
   p2r: false,
   horizontalScrolling: false,
   storage: true,
-  development: false,
+  development: true,
```
Comment (alabiaga) [inline src/framecontainer/viewerConfig.ts]: should this be defaulting to true? or did you mistakenly check this in when you changed it for your developing purposes?

**Comment by fstanis** [comment]
Closing in favor of #41

**Pull request closed without merging**